extended/src/main/java/io/kubernetes/client/extended/generic/GenericKubernetesApi.java

```diff
@@ -0,0 +1,680 @@
+        });
+  }
+
+  /**
+   * List kubernetes api response.
+   *
+   * @param namespace the namespace
+   * @param listOptions the list options
+   * @return the kubernetes api response
+   */
+  public KubernetesApiResponse<ApiListType> list(String namespace, final ListOptions listOptions) {
+    if (Strings.isNullOrEmpty(namespace)) {
+      throw new IllegalArgumentException("invalid namespace");
+    }
+    return executeCall(
+        customObjectsApi.getApiClient(),
+        apiListTypeClass,
+        () -> {
+          return customObjectsApi.listNamespacedCustomObjectCall(
+              this.apiGroup,
+              this.apiVersion,
+              namespace,
+              this.resourcePlural,
+              null,
+              listOptions.getContinue(),
+              listOptions.getFieldSelector(),
+              listOptions.getLabelSelector(),
+              listOptions.getLimit(),
+              listOptions.getResourceVersion(),
+              listOptions.getTimeoutSeconds(),
+              null,
+              null);
+        });
+  }
+
+  /**
+   * Create kubernetes api response.
+   *
+   * @param object the object
+   * @param createOptions the create options
+   * @return the kubernetes api response
+   */
+  public KubernetesApiResponse<ApiType> create(ApiType object, final CreateOptions createOptions) {
+    V1ObjectMeta objectMeta;
+    try {
+      objectMeta = Reflect.objectMetadata(object);
+    } catch (ObjectMetaReflectException e) {
+      throw new IllegalArgumentException("fail to extract object metadata");
+    }
+
+    return executeCall(
+        customObjectsApi.getApiClient(),
+        apiTypeClass,
+        () -> {
+          // TODO(yue9944882): judge namespaced object via api discovery
+          boolean isNamespaced = !Strings.isNullOrEmpty(objectMeta.getNamespace());
+          if (isNamespaced) {
+            return customObjectsApi.createNamespacedCustomObjectCall(
+                this.apiGroup,
+                this.apiVersion,
+                objectMeta.getNamespace(),
+                this.resourcePlural,
+                object,
+                null,
+                null);
+          } else {
+            return customObjectsApi.createClusterCustomObjectCall(
+                this.apiGroup, this.apiVersion, this.resourcePlural, object, null, null);
+          }
+        });
+  }
+
+  /**
+   * Update kubernetes api response.
+   *
+   * @param object the object
+   * @param updateOptions the update options
+   * @return the kubernetes api response
+   */
+  public KubernetesApiResponse<ApiType> update(ApiType object, final UpdateOptions updateOptions) {
+    V1ObjectMeta objectMeta;
+    try {
+      objectMeta = Reflect.objectMetadata(object);
+    } catch (ObjectMetaReflectException e) {
+      throw new IllegalArgumentException("fail to extract object metadata");
+    }
+    return executeCall(
+        customObjectsApi.getApiClient(),
+        apiTypeClass,
+        () -> {
```

Comment:
This function looks like it's duplicated with above? If so, pull out with a single implementation.

Reply:
the called function in the lambda has similar name but different:

replaceNamespacedCustomObjectCall vs replaceClusterCustomObjectCall